src/components/connectionWithWallet/ConnectWallet.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
+
+import { useNavigate } from "react-router-dom";
+
+const ConnectWallet: React.FC = () => {
+  const [connected, setConnected] = useState(false);
+  const [currentAccount, setCurrentAccount] = useState<string | null>(null);
+  const [balance, setBalance] = useState<string | null>(null);
+  const navigate = useNavigate();
+
+  const connectWallet = async () => {
+    if (window.ethereum) {
+      try {
+        if (connected) {
```

Comment:
what does this part do?
` if (connected) {
          setConnected(false);
          setCurrentAccount(null);
          setBalance(null);
        }`